gambatte_watchOS/GameCoreSnapshots.swift

```diff
@@ -60,4 +60,52 @@ extension GameCore {
 		}
 		return UIImage(cgImage: cgImage)
 	}
+	
+	public func createTexture(from buffer: UnsafeMutablePointer<UInt32>) -> SKTexture {
+		
+		let size = CGSize(width: GameplayController.screenWidth,
+		                  height: GameplayController.screenHeight)
+		let data = createData(from: buffer)
+		let texture = SKTexture(data: data, size: size, flipped: true)
+		texture.filteringMode = .nearest
+		return texture
+	}
+	
+	fileprivate func createData(from buffer: UnsafeMutablePointer<UInt32>) -> Data {
+		
+		let count = Int(kScreenHeight * kScreenWidth) * MemoryLayout<UInt32>.size
+		let bufferPointer =  UnsafeBufferPointer(start: buffer, count: count)
+		let data = Data(buffer: bufferPointer)
+		
+		var bytes = [UInt8](repeating: 0, count: count)
+		data.copyBytes(to: &bytes, count: count)
+		
+		var mdata = data
+		mdata.withUnsafeMutableBytes { (ptr: UnsafeMutablePointer<UInt8>) in
+			for i in stride(from: 0, to: count, by: 4) {
+				swap(&ptr[i], &ptr[i+2])
```

Comment:
Instead of swaping BGR to RGB at here every time when buffer updates, you might want to swap the color palette in [gambatte](https://github.com/gabrieloc/gambatte_watchOS/blob/cda024458a91c02174f21552ec3c98dfc1936240/libgambatte/src/video.cpp#L38).